### ISSUE TITLE
Web - Pay - Not here page opens when Gmail user Pay with Expensify

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49523]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.38-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** gocemate+a2207@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Open 1:1 chat with other user
2. Go to Pay
3. Select USD -$ currency>Enter amount> Next> Pay with Expensify

## Expected Result:
Add bank account page should open

## Actual Result:
Not here page opens when Gmail user Pay with Expensify

## Workaround:
Unknown

## Platforms:

- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b9c18590-c3bb-44f9-b8c3-e875c378dbb8

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by IuliiaHerets** [comment]
@isabelastisser  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by etCoderDysto** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Pay - Not here page opens when gmail user Pay with Expensify
### What is the root cause of that problem?
When user is not validated and navigates to `bankAccountRoute` (settings/wallet/add-bank-account) hmm it not page is displayed 
https://github.com/Expensify/App/blob/d009d8ed4830c7d0562c524c991b8a688e2b4869/src/components/MoneyRequestConfirmationList.tsx#L849

https://github.com/Expensify/App/blob/d009d8ed4830c7d0562c524c991b8a688e2b4869/src/pages/AddPersonalBankAccountPage.tsx#L70

### What changes do you think we should make in order to solve the problem?
We should check if the user is validated first before we display pay with expensify button 
* get currentUserPersoanlDetails using 
```
const currentUserPersonalDetails = useCurrentUserPersonalDetails();
```
* Display paywith Expensify button when the user is validated 
```
const canUseWallet = !isExpenseReport && !isInvoiceReport && currency === CONST.CURRENCY.USD && currentUserPersonalDetails.validated;
```


https://github.com/Expensify/App/blob/d009d8ed4830c7d0562c524c991b8a688e2b4869/src/components/SettlementButton/index.tsx#L98
### What alternative solutions did you explore? (Optional)

**Comment by isabelastisser** [comment]
@davidcardoza, I can't add this issue to the `bill pay` project. Can you please help? Thanks!

**Comment by 1subodhpathak** [comment]
Not able to reproduce!

**Comment by ikevin127** [comment]
♻️ Reviewing proposals in progress.

**Comment by ikevin127** [comment]
🎉 Thanks everybody for the proposals and great variety in solutions!

##

From reviewing the existing proposals, three variants are surfacing (see below) and I'm not sure whether this is a design or business decision but I'll tag @Expensify/design anyways to get their take on this before assignment since the team was active recently in PR #49230.

ℹ️ As reviewer, I lean towards **proposal 3** because, given PR #49230, we can agree that some work was put into this flow already specifically for the case when user account is not validated yet. My second option would be **proposal 2** since it's the closest to what *I think* will be preferred here, and we wouldn't need to HOLD if we were to go with this version.

| Proposal 1: ♻️ Hide `Pay with Expensify` button via `canUseWallet` when using unvalidated account, ⏳ Can be implemented right away | Proposal 2: ♻️ Display `ValidateAccountMessage` component when pressing `Pay with Expensify` with unvalidated account, ⏳ Can be implemented right away | Proposal 3 [HOLD #49230]: ♻️ Display new `VERIFY_ACCOUNT` screen when pressing `Pay with Expensify` with unvalidated account, ⚠️ Can be implemented as soon as https://github.com/Expensify/App/pull/49230 is merged |
| ------------- | ------------- | ------------- |
| <img width="300" alt="p1" src="https://github.com/user-attachments/assets/ccda3b28-fcde-458f-8a79-1926693534ab"> | <img width="300" alt="p2" src="https://github.com/user-attachments/assets/a16143ff-91ba-4820-ae58-e314a1cb6c7a"> | <img width="300" alt="p3" src="https://github.com/user-attachments/assets/f719caae-3a07-4d1e-8b38-d25bc2a932b9"> |

**Comment by dubielzyk-expensify** [comment]
Yeah I kinda wanna hear what @JmillsExpensify and @trjExpensify thinks too. I can see an argument for both 1 and 3. 2 feels weird to me

**Comment by dannymcclain** [comment]
Agree that I'd like to hear from Jason and Tom, but my understanding was that Proposal 3 was how we wanted to handle all these unvalidated account situations (basically just add the validate screen to the beginning of whatever flow the user was trying to go through) so I have a strong preference for that option.

**Comment by trjExpensify** [comment]
Yep, same. Proposal 3 for the win. 

I'd also say let's make sure that copy gives them a hint to go check their mail to grab the magic code we just fired off. I.e "This feature requires you to validate your account. Enter the magic code sent to your email."

**Comment by ikevin127** [comment]
Based on the latest discussions looks like @cretadn22's [proposal](https://github.com/Expensify/App/issues/49523#issuecomment-2364278463) is the way to go here. The RCA is correct and the proposed solution was verified by the design team.

**Note**: Even though we assign the contributor now, we should HOLD https://github.com/Expensify/App/pull/49230 until the PR is merged since the proposed solution is dependent on that PRs additions.

🎀👀🎀 C+ reviewed

**Comment by ikevin127** [comment]
> Even though we assign the contributor now, we should [HOLD #49230] until the PR is merged since the assigned solution is dependent on that PRs additions.

**Comment by ikevin127** [comment]
ℹ️ **Status update**:
- PR https://github.com/Expensify/App/pull/49230 was merged which means we can move forward with our [selected proposal](https://github.com/Expensify/App/issues/49523#issuecomment-2374749015) solution and open the PR with the fix for this issue

- ♻️ PR https://github.com/Expensify/App/pull/50169 is in review

**Comment by isabelastisser** [comment]
@ikevin127 @cretadn22, can you please provide an update? Did the PR cause a regression? What's next here?

**Comment by ikevin127** [comment]
@isabelastisser Yes, we had the following regression:
- https://github.com/Expensify/App/issues/50284

which was caused by lack of testing on both PR reviewer and author sides. The issue mentioned above is open for proposals, unless the author wants to fix the issue given that we're still within the regression period.

**Note**: The issue was that we relied on the other PRs implementation of the new route without actually testing if the implementation works, ignoring this issue's **Expected result** which states `Add bank account page should open` as we stopped the testing at "seeing the magic code input showing up in RHP".

**Comment by ikevin127** [comment]
@cretadn22 No. Truth is that because neither of us actually tested what happens once the validation code is inputted - we're both going to get the regression penalty here.

It's also true that the issue of being logged out was not caused directly by logic you introduced in the PR, but rather an edge case / side effect of using the logic of the other PR in our specific case.

Given this, as I mentioned in the regression issue -> it's open for proposals.

**Comment by ikevin127** [comment]
* [x]  [@ikevin127] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/37174.
* [x]  [@ikevin127] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/37174/files#r1797481938.
* [x]  [@ikevin127] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**.
* [x]  [@ikevin127] Determine if we should create a regression test for this bug.
* [x]  [@ikevin127] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal

1. Login to new account (unverified).
2. Start a chat with an existing user > [+] > Pay [userName].
3. Select USD > Pay with Expensify.
4. Verify that the 'verify account page' shows up in RHP prompting user to enter the magic code.

Do we agree 👍 or 👎.

##

@isabelastisser As for payment, there was one regression:
- https://github.com/Expensify/App/issues/50284

which wasn't caused directly by the code implemented by this issue's fix but as a side effect of our previous logic not being adjusted to handle updating the `authToken` while logged in - more details in issue https://github.com/Expensify/App/issues/50284#issuecomment-2408136449.

Given this, both me (reviewer) and contributor should get a -50% penalty on the issue's bounty.

**Comment by isabelastisser** [comment]
I was OOO yesterday. I will review this today!

**Comment by isabelastisser** [comment]
@ikevin127, thanks for the detailed explanation above! I've processed the payments in Upwork now.
